tests/src/env.cpp

```diff
@@ -0,0 +1,32 @@
+#include <catch2/catch.hpp>
+
+/*
+ * Catch2: examples
+ * for more examples checkout:
+ *   https://github.com/catchorg/Catch2/tree/devel/examples
+ *
+ * Catch has two natural expression assertion macro's:
+ * - REQUIRE() stops at first failure.
+ * - CHECK() continues after failure.
+ *
+ * There are two variants to support decomposing negated expressions:
+ * - REQUIRE_FALSE() stops at first failure.
+ * - CHECK_FALSE() continues after failure.
+ *
+ * TEST_CASE( "Title here", "[short name]" ) {
+ *     REQUIRE( [BOOLEAN CONDITION] ) ;
+ * }
+ */
+
+extern "C" {
+# include "env.h"
+}
+
+#include <stdio.h>
+
+TEST_CASE( "create_destroy env", "[env]" ) {
+	t_env *env = env_create();
+	REQUIRE(env != NULL);
+	env_destroy(env);
+}
+
```

Comment:
good starting point. make sure to test the shit out of the environment stuff, since we are going to use it all over the place and bugs could be very hard to spot when the parser get's involved with stuff